bootstrap-languages/language-language/hc-dna/zomes/language_storage/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use integrity::LinkTypes;
+use hdk::prelude::*;
+
+pub(crate) fn err(reason: &str) -> WasmError {
+    wasm_error!(WasmErrorInner::Host(String::from(reason)))
+}
+
+pub(crate) fn get_latest_link(base: EntryHash, tag: Option<LinkTag>) -> ExternResult<Option<Link>> {
+    let language_info = get_links(base, LinkTypes::LanguageLink, tag)?;
+
+    // Find the latest
+    let latest_info =
+        language_info
+            .into_iter()
+            .fold(None, |latest: Option<Link>, link| match latest {
+                Some(latest) => {
+                    if link.timestamp > latest.timestamp {
```

Comment:
Yeah, might be easy to change this to `get_oldest_link` like this.
```suggestion
                    if link.timestamp < latest.timestamp {
```

Reply:
And we should think about enabling multiple LanguageMetas (just keeping the first as the normative one) so that multiple agents actually can sign a language and the language controller iterates through all signatures/metas from these different agents if it can't establish a trusted signature.